pandas/io/excel/_base.py

```diff
@@ -105,25 +105,24 @@
     Supported engines: "xlrd", "openpyxl", "odf", "pyxlsb".
     Engine compatibility :
 
-    - "xlrd" supports most old/new Excel file formats.
+    - "xlrd" supports old-style Excel files (.xls).
     - "openpyxl" supports newer Excel file formats.
     - "odf" supports OpenDocument file formats (.odf, .ods, .odt).
     - "pyxlsb" supports Binary Excel files.
 
     .. versionchanged:: 1.2.0
         The engine `xlrd <https://xlrd.readthedocs.io/en/latest/>`_
-        is no longer maintained, and is not supported with
-        python >= 3.9. When ``engine=None``, the following logic will be
-        used to determine the engine.
+        now only supports old-style ``.xls`` files.
+        When ``engine=None``, the following logic will be
+        used to determine the engine:
 
         - If ``path_or_buffer`` is an OpenDocument format (.odf, .ods, .odt),
           then `odf <https://pypi.org/project/odfpy/>`_ will be used.
-        - Otherwise if ``path_or_buffer`` is a bytes stream, the file has the
+        - Otherwise if the file has the
           extension ``.xls``, or is an ``xlrd`` Book instance, then ``xlrd`` will
           be used.
         - Otherwise if `openpyxl <https://pypi.org/project/openpyxl/>`_ is installed,
           then ``openpyxl`` will be used.
-        - Otherwise ``xlrd`` will be used and a ``FutureWarning`` will be raised.
```

Comment:
I have not made the code changes related to this line or the change on line 121 (these changes are echoed in the section around line ~900), but that's what I think the approach should be.

Reply:
Well, so that will be the approach in the future. But in pandas we decided to first raise a warning about it (in case the user only upgraded pandas, and not xlrd). 
So for now we should keep this sentence (but maybe we should add something like "... and this ability will be removed in a future version" to make this more explicit)

Reply:
While I respect the choice of approach, I can't really condone it. As I said above: I have concerns around security with xlrd<2, and so anything that can be done to stop people using it would be a very good thing.

If a user is upgrading such that they have pandas 1.2, that would be a very good time for them to also upgrade to xlrd 2 and whatever the latest openpyxl release is...

Reply:
> I have not made the code changes related to this..., but that's what I think the approach should be.

Just so I understand the intentions here, are you planning to make those code changes (taking into account @jorisvandenbossche request)?

Reply:
It's an open question, see https://github.com/pandas-dev/pandas/issues/38424#issuecomment-744062773